src/parser.c

```diff
@@ -130,32 +124,40 @@ void handleMarkdownStyleLink(char buf[1000], slide *s) {
                 }
             }
 
-            // create the url
-            currentPosition = 0;
-            char urlBuffer[1000];
-            do {
-                urlBuffer[currentPosition] = *(urlStart+currentPosition);
-                currentPosition++;
-            } while (!(urlStart+currentPosition>urlEnd));
-            urlBuffer[currentPosition]='\0';
-
-            // create link
-            mdlink *l;
-            if (s->link !=NULL) {
-                l = appendLink(s->link);
+            size_t parsedURLLength = urlEnd - urlStart + 1;
+            char *url;
+            size_t urlLength;
+            bool shouldFreeURL;
+            // make sure that http in url before assigning as value to struct
+            if (strnstr(urlStart, "http://", parsedURLLength) || 
```

Comment:
`strnstr` is giving me the following error when I run `make install`:

'''
/usr/bin/ld: src/parser.o: in function `handleMarkdownStyleLink':
parser.c:(.text+0x273): undefined reference to `strnstr'
/usr/bin/ld: parser.c:(.text+0x2de): undefined reference to `strnstr'
collect2: error: ld returned 1 exit status
make: *** [Makefile:39: dss] Error 1
'''

I did a quick google on this. Looks like it is part of the FreeBSD C implementation, but not in the standard library?

Do you know of an alternative function we could use here? perhaps `strstr`?

Reply:
I was afraid of that.  I knew it wasn't POSIX but was hoping it would be more widely available.   Unfortunately `strstr` won't quite work because we don't want to match things like

```
[title](www.site.com) http://www.another.org/
```

`strnstr` was helping out by requiring that `http://` be within the url itself, not just somewhere after its start in the line.

One solution would be to add an implementation of `strnstr`.  

In this case, though, it might make more sense to change the problem that we're solving a bit: rather than requiring that the url contain `http://` or `https://` *somewhere*, we could require that the url *start with* one of those two.  I added a small change to check whether the url begins with one of those: https://github.com/codythegreat/dss/pull/50/commits/1180dedce6f33fe78d0063c8b7a362ea909a35e2 .

Reply:
> In this case, though, it might make more sense to change the problem that we're solving a bit: rather than requiring that the url contain http:// or https:// somewhere, we could require that the url start with one of those two. I added a small change to check whether the url begins with one of those: 1180ded .

I like this approach. No need to reinvent the wheel if we can just make a little tweak to our problem.

I'll review the change and if all goes well I should be able to merge your changes soon.